idseq_dag/steps/run_lzw.py

```diff
@@ -44,20 +63,52 @@ def lzw_fraction(sequence):
         return float(len(results)) / len(sequence)
 
     @staticmethod
-    @command.run_in_subprocess
-    def generate_lzw_filtered(fasta_files, output_files, cutoff_fractions):
+    def lzw_compute(input_files, slice_step=NUM_SLICES, lzw_fraction=lzw_fraction):
+        """Spawn subprocesses on NUM_SLICES of the input files, then coalesce the
+        scores into a temp file, and return that file's name."""
+
+        temp_file_names = ["lzwslice_{slice_step}_{slice_start}.txt" for slice_start in range(slice_step + 1)]
+        for tfn in temp_file_names:
+            assert not os.path.exists(tfn)
```

Comment:
Remember that asserts are stripped if python is started with `-O`...

Reply:
Yep, but nobody does that, since it doesn't produce any speed up :P)

Good observation, though.   In past I've used a function called "reqiuire" with the same semantics as "assert".   Should I add that and change all the asserts to requires?

Reply:
That's very cool, I didn't know about "x" mode.   Glad to take advantage of more Python-3 goodness.   Cleanup is not an issue at all in this case;  I am replacing a failed assert exception with another exception :)

Note, btw, that there are no problems with race conditions in this use case.   The subprocesses operate on different files.

Reply:
Ok, and this is the only process running on that machine ?

Reply:
AFAIK this is meant to be the only job running inside a docker container, and runs inside a directory that is specific to this run.   @yunfangjuan can let us know if my assumptions are too strong, and in that case i could make the names unique so they don't clash.   But seems to me like the uniqueness ought to be assured at the higher level (e.g. the directory is unique to each run).

Reply:
Sounds good.

Reply:
I looked into "x" mode some more, and it probably uses some filesystem locking to ensure exclusive creation, which makes me a bit less enthusiastic to use it.  Besides, we don't actually expect concurrent processes to create the same file;  that assert is only meant to fire if you are a developer mindlessly rerunning something in the same folder.

Reply:
Not sure what kind of locking you are referring too that would not be use later on when you actually create the file. 
In any case, as you said, given all other assumptions and guarantees, this is a very minor clean up.